authentication.go

```diff
@@ -178,3 +204,75 @@ func (s *AuthenticationService) GetCurrentUser() (*Session, error) {
 
 	return ret, nil
 }
+
+// Attempts to retrieve an Oauth2 access token from the Jira Auth Server. Returns the details of the access token
+// in a JiraAccessToken struct so that it can be cached by the client application if desired (oauth2 tokens expire
+// after 15 min by default); else, this can be called multiple times to retrieve a new access token.
+//
+// oauthClientId, userKey, and sharedSecret are returned by Jira during the installation of the Add-on
+// scope is the scope to place on the AccessToken returned (e.g., "READ", "READ WRITE", "READ ACT_AS_USER")
+// See: https://developer.atlassian.com/cloud/jira/software/oauth-2-jwt-bearer-token-authorization-grant-type
+func (s *AuthenticationService) GetOauth2AccessToken(oauthClientId, userKey, scope string, sharedSecret []byte) (*JiraAccessToken, error) {
+	// create JSON Web Token
+	jiraAuthURL := "https://auth.atlassian.io"
```

Comment:
This seems like it should be:
1. A constant at the top of the `AuthenticationService`
2. A config variable with a default

I'm not sure how many oauth2 services that jira will possibly expect.  That would inform which one you'd pick, I think.

Reply:
After looking at the rest of the PR, it looks like this whole thing is very closely tied to the JIRA-specific oauth2 service, so probably option 1 above.